### PR TITLE
[pipelines] Add support for global values and postgresql 12.x

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
[pipelines] Add support for global values and postgresql 12.x

* **Breaking change:** Updated `imagePullSecrets` value from string to list.
* Added support for global values
* Updated maintainers in chart.yaml
* Update postgresql tag version to `12.3.0-debian-10-r71`
* Update postgresql sub chart version to `9.3.4` - [9.x Upgrade Notes](https://github.com/bitnami/charts/tree/master/bitnami/postgresql#900)
* Update rabbitmq sub chart version to `7.4.3` - [7.x Upgrade Notes](https://github.com/bitnami/charts/tree/master/bitnami/rabbitmq#to-700)
* Please refer [here](https://github.com/jfrog/charts/blob/master/stable/pipelines/README.md#special-upgrade-notes) for upgrade notes
* Update redis sub chart version to `10.9.0`
* **IMPORTANT**
* If this is a new deployment or you already use an external database (`postgresql.enabled=false`), these changes **do not affect you**!
* If this is an upgrade and you are using the default PostgreSQL (`postgresql.enabled=true`), you need to pass previous 9.x's postgresql.image.tag and databaseUpgradeReady=true

### CODE REVIEW

**Review by danielezer — APPROVED** [review]

File: stable/pipelines/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: pipelines
 home: https://jfrog.com/pipelines/
-version: 1.4.3
+version: 2.0.0
```
Comment (rimusz) [inline stable/pipelines/Chart.yaml:4]: Please change version to patch 1.4.4

File: stable/pipelines/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,17 @@
 # JFrog Pipelines Chart Changelog
 All changes to this chart to be documented in this file
 
+## [2.0.0] Sept 4, 2020
+* **Breaking change:** Modified `imagePullSecrets` value from string to list.
+* Added support for global values
+* Updated maintainers in chart.yaml
+* Update postgresql tag version to `12.3.0-debian-10-r71`
+* Update postgresql sub chart version to `9.3.4` - [9.x Upgrade Notes](https://github.com/bitnami/charts/tree/master/bitnami/postgresql#900)
+* Removed redundant mcKey
```
Comment (rimusz) [inline stable/pipelines/CHANGELOG.md]: What is this? there is/was no mcKey

File: stable/pipelines/values.yaml [hunk 3]
```diff
@@ -425,10 +464,10 @@ pipelines:
       artifactory:
         ## Artifactory URL
         ##
-        baseUrl: "{{ tpl (required "\n\npipelines.jfrogUrl is required!\n" .Values.pipelines.jfrogUrl) . }}"
+        baseUrl: "{{ tpl (required "\n\npipelines.jfrogUrl or global.jfrogUrl is required! This allows to connect to Artifactory.\nYou can copy the JFrog URL from Admin > Security > Settings" (include "pipelines.jfrogUrl" .)) . }}"
         ## Unified UI URL
         ##
-        baseUrlUI: "{{ tpl (required "\n\npipelines.jfrogUrlUI is required!\n" .Values.pipelines.jfrogUrlUI) . }}"
+        baseUrlUI: "{{ tpl (required "\n\npipelines.jfrogUrlUI is required!" (include "pipelines.jfrogUrl" .)) . }}"
```
Comment (rimusz) [inline stable/pipelines/values.yaml:470]: Why it got replaced with `jfrogUrl` it must stay `jfrogUrlUI`.

File: stable/pipelines/values.yaml [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+    port:
+    token:
+    ## Vault token using existing secret
+    # existingSecret: secret
+
+
+  # imagePullSecrets:
+  #   - myRegistryKeySecretName
+  # jfrogUrl:
+  # joinKey:
```
Comment (rimusz) [inline stable/pipelines/values.yaml:29]: Add back `jfrogUrlUi:`

File: stable/pipelines/templates/_helpers.tpl [hunk 5]
```diff
@@ -114,3 +114,99 @@ Set grcp url
 {{- printf "%s" (tpl .Values.pipelines.jfrogUrl . ) }}
 {{- end }}
 {{- end -}}
+
+{{/*
+Resolve jfrogUrl value
```
Comment (rimusz) [inline stable/pipelines/templates/_helpers.tpl:119]: Add the same for `jfrogUrlUi` as well

**Review by rimusz — CHANGES_REQUESTED** [review]

File: stable/pipelines/Chart.yaml (hunk 1 above)
Comment (chukka) [reply stable/pipelines/Chart.yaml:4]: @rimusz  This is a breaking change for imagePull secrets  and upgrade to 12.x postgresql - Hence added Major version bump

File: stable/pipelines/values.yaml (hunk 4 above)
Comment (chukka) [reply stable/pipelines/values.yaml:29]: Will add

File: stable/pipelines/templates/_helpers.tpl (hunk 5 above)
Comment (chukka) [reply stable/pipelines/templates/_helpers.tpl:119]: Will add

File: stable/pipelines/Chart.yaml (hunk 1 above)
Comment (rimusz) [reply stable/pipelines/Chart.yaml:4]: ok

File: stable/pipelines/values.yaml (hunk 4 above)
Comment (chukka) [reply stable/pipelines/values.yaml:30]: Done

File: stable/pipelines/values.yaml [hunk 6]
```diff
@@ -623,7 +648,7 @@ pipelines:
       installerIP: "1.2.3.4"
       installerAuthToken: "{{ .Values.pipelines.authToken }}"
       installerImage: "jfrog/pipelines-installer"
-      registryUrl: "{{ include "pipelines.imageRegistry" . }}"
+      registryUrl: "{{ .Values.global.imageRegistry }}"
```
Comment (chukka) [inline stable/pipelines/values.yaml:651]: @rahulsadanandan  please check  if this is used anywhere in pipelines installer ?

Comment (rahulsadanandan) [reply stable/pipelines/values.yaml:651]: Not sure where this is used , 
@jatindogra @rimusz Can we hardcode this value in systemYaml to `docker.bintray.io` like it is used in 
`languageImages:` section

File: stable/pipelines/values.yaml [hunk 7]
```diff
@@ -138,7 +142,7 @@ pipelines:
 
   api:
     image:
-      registry: docker.bintray.io
+      # registry:
```
Comment (rimusz) [inline stable/pipelines/values.yaml:145]: why this was added?

Comment (chukka) [reply stable/pipelines/values.yaml:145]: we made this change to sync with all charts (jfrog charts and external charts)

**Comment by chukka** [comment]
closing this - opening #1169

**Pull request closed without merging**